### PR TITLE
Es/fatal error

### PR DESCRIPTION
**Pull request by ErikSin**
Shows an error page when server has incorrectly started.

image

### CODE REVIEW

File: src/frontend/ServerLoading.tsx [hunk 1]
```diff
@@ -38,12 +40,8 @@ export const ServerLoading = ({
...
 
   if (serverStatus.value === 'ERROR') {
-    return (
-      <SafeAreaView
-        style={{flex: 1, justifyContent: 'center', alignItems: 'center'}}>
-        <Text>Error!</Text>
-      </SafeAreaView>
-    );
+    BootSplash.hide();
+    return <FatalError />;
```
Comment (ErikSin) [inline src/frontend/ServerLoading.tsx:44]: We could possibly put this in an error boundary, but I am unsure of the best practices with fatal errors, so im curious what you think @achou11 .

All that being said, even if this was encapsulated in an error boundary, I think we still would need to put this `FatalError` Component here as well. `BootSplash.hide()` is async, so throwing an error after `hide()` would not get caught by the ErrorBoundary, as they are only captured within the react render cycle

File: src/frontend/ServerLoading.tsx [hunk 2]
```diff
@@ -12,6 +13,7 @@ export const ServerLoading = ({
   const [serverStatus, setServerStatus] = React.useState<StatusMessage>({
     value: 'STARTING',
   });
+  console.log(serverStatus);
```
Comment (achou11) [inline src/frontend/ServerLoading.tsx:16]:
```suggestion
```

File: src/frontend/ServerLoading.tsx (hunk 1 above)
Comment (achou11) [reply src/frontend/ServerLoading.tsx:44]: i think it makes sense here. the error boundary is more for react rendering errors that come up, whereas this is capturing an error happening outside of react (the server). mapeo mobile essentially does what you're doing here, if that's any confirmation 😄

**Review by achou11 — APPROVED** [review]
aside from a stray console log, lgtm!

File: src/frontend/screens/FatalError.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            height={50}
+            style={{position: 'absolute', bottom: 0, right: 0}}
+          />
+        </View>
+        <Text style={styles.text}>{formatMessage(m.somethingWrong)}</Text>
+      </View>
+      <Button
+        fullWidth
+        onPress={() => {
+          RNRestart.restart();
```
[the remark this answers is not in the document]
Comment (achou11) [reply src/frontend/screens/FatalError.tsx:41]: i think it should suffice, as long as that means that `nodejs.start()` is being called again. hard to know without testing though...

[the remark this answers is not in the document]
Comment (achou11) [reply src/frontend/screens/FatalError.tsx:41]: the safe alternative is to just provide copy about closing and re-opening the app

**Comment by achou11** [comment]
hm looking into the restart dep and starting to have questions about whether the app bundle restart is enough in the case of a server crash 🤔 ideally can confirm that it works by throwing an error in the backend startup script and testing the button, but haven't tried that myself yet.

Can confidently say that the UI implemented here lgtm, but yeah just some questions around how that restart button actually works 🤷

**Pull request merged**